framework/validators/UniqueValidator.php

```diff
@@ -106,7 +106,7 @@ public function validateAttribute($model, $attribute)
             $query->andWhere($this->filter);
         }
 
-        if (!$model instanceof ActiveRecordInterface || $model->getIsNewRecord() || $model->className() !== $targetClass) {
+        if (!$model instanceof ActiveRecordInterface || $model->getIsNewRecord() || $model->className() !== $targetClass::className()) {
```

Comment:
`::className()` is a the most reliable way, as for me. Since we load the class on line 100, extra `className()` does not impact the performance.